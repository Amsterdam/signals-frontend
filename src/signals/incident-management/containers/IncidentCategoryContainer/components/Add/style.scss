.incident-category-add {
  .action {
    height: 52px;
  }
}
