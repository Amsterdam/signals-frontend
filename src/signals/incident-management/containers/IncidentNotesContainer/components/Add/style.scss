.incident-notes-add {
  &__submit {
    margin-top: 16px;
  }
}
