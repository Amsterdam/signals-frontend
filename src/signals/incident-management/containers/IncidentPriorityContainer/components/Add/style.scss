.incident-priority-add {
  .action {
    height: 52px;
  }
}
