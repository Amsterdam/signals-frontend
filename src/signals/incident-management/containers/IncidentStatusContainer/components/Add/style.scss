.incident-status-add {
  &__submit {
    margin-top: 16px;
  }
}
