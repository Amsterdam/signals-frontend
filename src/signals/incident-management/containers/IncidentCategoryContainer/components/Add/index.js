import React from 'react';
import PropTypes from 'prop-types';
import { FormBuilder, FieldGroup, Validators } from 'react-reactive-form';

import FieldControlWrapper from '../../../../components/FieldControlWrapper';
import SelectInput from '../../../../components/SelectInput';
import './style.scss';


class Add extends React.Component { // eslint-disable-line react/prefer-stateless-function
  categoryForm = FormBuilder.group({ // eslint-disable-line react/sort-comp
    _signal: [''],
    sub: ['', Validators.required],
    loading: false
  });

  handleSubmit = (event) => {
    event.preventDefault();
    const status = { ...this.categoryForm.value, _signal: this.props.id };
    this.props.onRequestCategoryUpdate(status);
  }

  componentWillUpdate(props) {
    if (props.loading !== this.props.loading) {
      this.categoryForm.controls.loading.setValue(props.loading);
    }
  }

  render() {
    const { subcategoryList, loading } = this.props;
    return (
      <div className="incident-category-add">
        <div className="incident-category-add__body">
          <FieldGroup
            control={this.categoryForm}
            render={({ invalid }) => (
              <form onSubmit={this.handleSubmit}>
                <div>
                  <FieldControlWrapper
                    render={SelectInput}
                    name="sub"
                    display="Subcategorie"
                    control={this.categoryForm.get('sub')}
                    values={subcategoryList}
                    multiple={false}
                    emptyOptionText="Selecteer..."
                  />

                  <button className="action primary" type="submit" disabled={invalid || loading}>
                    <span className="value">Subcategorie wijzigen</span>
                    {loading ?
                      <span className="working">
                        <div className="progress-indicator progress-white"></div>
                      </span>
                    : ''}
                  </button>
                </div>
              </form>
            )}
          />
        </div>
      </div>
    );
  }
}

Add.defaultProps = {
  loading: false
};

Add.propTypes = {
  id: PropTypes.string,
  subcategoryList: PropTypes.array,
  loading: PropTypes.bool,

  onRequestCategoryUpdate: PropTypes.func.isRequired
};

export default Add;
