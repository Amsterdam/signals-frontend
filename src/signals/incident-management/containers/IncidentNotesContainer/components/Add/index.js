import React from 'react';
import PropTypes from 'prop-types';
import { FormBuilder, FieldGroup, Validators } from 'react-reactive-form';

import FieldControlWrapper from '../../../../components/FieldControlWrapper';
import TextAreaInput from '../../../../components/TextAreaInput';
import './style.scss';


class Add extends React.Component { // eslint-disable-line react/prefer-stateless-function
  noteForm = FormBuilder.group({ // eslint-disable-line react/sort-comp
    _signal: [''],
    text: [null, Validators.required],
    loading: false
  });

  componentWillUpdate(props) {
    if (props.loading !== this.props.loading) {
      this.noteForm.controls.loading.setValue(props.loading);
    }
  }

  handleSubmit = (event) => {
    event.preventDefault();
    const note = { ...this.noteForm.value, _signal: this.props.id };
    this.props.onRequestNoteCreate(note);
    this.noteForm.reset();
  }

  render() {
    const { error, loading } = this.props;
    return (
      <div className="incident-notes-add">
        <div className="incident-notes-add__body">
          <FieldGroup
            control={this.noteForm}
            render={({ invalid }) => (
              <form onSubmit={this.handleSubmit}>
                <div>
                  <FieldControlWrapper render={TextAreaInput} name="text" display="Notitie" control={this.noteForm.get('text')} rows={5} />

                  {error ? <div className="notification notification-red" >De gekozen notitie is niet mogelijk in deze situatie.</div> : ''}

                  <button className="incident-notes-add__submit action primary" type="submit" disabled={invalid || loading}>
                    <span className="value">Notitie toevoegen</span>
                    {loading ?
                      <span className="working">
                        <div className="progress-indicator progress-white"></div>
                      </span>
                    : ''}
                  </button>
                </div>
              </form>
            )}
          />
        </div>
      </div>
    );
  }
}

Add.defaultProps = {
  loading: false,
  loadingExternal: false,
  error: false
};

Add.propTypes = {
  id: PropTypes.string,
  loading: PropTypes.bool,
  error: PropTypes.oneOfType([PropTypes.bool, PropTypes.object]),

  onRequestNoteCreate: PropTypes.func.isRequired
};

export default Add;
