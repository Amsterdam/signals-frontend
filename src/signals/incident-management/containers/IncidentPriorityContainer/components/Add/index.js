import React from 'react';
import PropTypes from 'prop-types';
import { FormBuilder, FieldGroup, Validators } from 'react-reactive-form';

import FieldControlWrapper from '../../../../components/FieldControlWrapper';
import SelectInput from '../../../../components/SelectInput';
import './style.scss';


class Add extends React.Component { // eslint-disable-line react/prefer-stateless-function
  priorityForm = FormBuilder.group({ // eslint-disable-line react/sort-comp
    _signal: [''],
    priority: ['', Validators.required],
    loading: false
  });

  handleSubmit = (event) => {
    event.preventDefault();
    const status = { ...this.priorityForm.value, _signal: this.props.id };
    this.props.onRequestPriorityUpdate(status);
  }

  componentWillUpdate(props) {
    if (props.loading !== this.props.loading) {
      this.priorityForm.controls.loading.setValue(props.loading);
    }
  }

  render() {
    const { priorityList, loading } = this.props;
    return (
      <div className="incident-priority-add">
        <div className="incident-priority-add__body">
          <FieldGroup
            control={this.priorityForm}
            render={({ invalid }) => (
              <form onSubmit={this.handleSubmit}>
                <div>
                  <FieldControlWrapper
                    render={SelectInput}
                    name="priority"
                    display="Urgentie"
                    control={this.priorityForm.get('priority')}
                    values={priorityList}
                    multiple={false}
                    emptyOptionText="Selecteer..."
                  />

                  <button className="action primary" type="submit" disabled={invalid || loading}>
                    <span className="value">Urgentie wijzigen</span>
                    {loading ?
                      <span className="working">
                        <div className="progress-indicator progress-white"></div>
                      </span>
                    : ''}
                  </button>
                </div>
              </form>
            )}
          />
        </div>
      </div>
    );
  }
}

Add.defaultProps = {
  loading: false
};

Add.propTypes = {
  id: PropTypes.string,
  priorityList: PropTypes.array,
  loading: PropTypes.bool,

  onRequestPriorityUpdate: PropTypes.func.isRequired
};

export default Add;
