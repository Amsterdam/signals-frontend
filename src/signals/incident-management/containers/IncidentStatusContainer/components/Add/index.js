import React from 'react';
import PropTypes from 'prop-types';
import { FormBuilder, FieldGroup, Validators } from 'react-reactive-form';

import FieldControlWrapper from '../../../../components/FieldControlWrapper';
import TextAreaInput from '../../../../components/TextAreaInput';
import SelectInput from '../../../../components/SelectInput';
import './style.scss';


class Add extends React.Component { // eslint-disable-line react/prefer-stateless-function
  // constructor(props) {
    // super(props);
    // this.sendToSigmax = this.sendToSigmax.bind(this);
  // }

  statusForm = FormBuilder.group({ // eslint-disable-line react/sort-comp
    _signal: [''],
    state: ['', Validators.required],
    text: [''],
    loading: false
  });

  componentWillUpdate(props) {
    if (props.loading !== this.props.loading) {
      this.statusForm.controls.loading.setValue(props.loading);
    }
  }

  handleSubmit = (event) => {
    event.preventDefault();
    const status = { ...this.statusForm.value, _signal: this.props.id };
    this.props.onRequestStatusCreate(status);
    this.statusForm.reset();
  }

  // sendToSigmax = () => {
    // const status = { _signal: this.props.id, state: 'i', text: 'sigmax' };
    // this.props.onRequestStatusCreate(status);
  // }

  render() {
    // TODO enable this when sigmax link is implemented
    // const { incidentStatusList, statusList } = this.props;
    // const currentState = incidentStatusList[incidentStatusList.length - 1].state;
    // const canSendToSigmax = !['i', 'o', 'a'].some((value) => value === currentState);
    const { statusList, loading, error } = this.props;
    // const canSendToSigmax = false; // disabled for the moment
    return (
      <div className="incident-status-add">
        {/* <div className="incident-status-add__title">Status toevoegen</div> */}
        <div className="incident-status-add__body">
          <FieldGroup
            control={this.statusForm}
            render={({ invalid }) => (
              <form onSubmit={this.handleSubmit}>
                <div>
                  <FieldControlWrapper render={SelectInput} name="state" display="Status" control={this.statusForm.get('state')} values={statusList} multiple={false} emptyOptionText="Selecteer..." />
                  <FieldControlWrapper render={TextAreaInput} name="text" display="Omschrijving" control={this.statusForm.get('text')} rows={5} />

                  {error ? <div className="notification notification-red" >De gekozen status is niet mogelijk in deze situatie.</div> : ''}

                  <button className="incident-status-add__submit action primary" type="submit" disabled={invalid || loading}>
                    <span className="value">Status toevoegen</span>
                    {loading ?
                      <span className="working">
                        <div className="progress-indicator progress-white"></div>
                      </span>
                    : ''}
                  </button>
                  <div>
                  </div>
                </div>
              </form>
            )}
          />
        </div>
      </div>
    );
  }
}

/*
{canSendToSigmax ?
  <button className="action tertiair" type="button" onClick={this.sendToSigmax}>
    <span className="value">Naar sigmax sturen</span>
  </button> : ''
}
*/

Add.defaultProps = {
  loading: false,
  error: false
};

Add.propTypes = {
  id: PropTypes.string,
  statusList: PropTypes.array,
  loading: PropTypes.bool,
  error: PropTypes.oneOfType([PropTypes.bool, PropTypes.object]),

  onRequestStatusCreate: PropTypes.func.isRequired
};

export default Add;
