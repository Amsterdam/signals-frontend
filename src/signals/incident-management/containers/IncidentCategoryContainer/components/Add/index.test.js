import React from 'react';
import { shallow } from 'enzyme';
import { FieldGroup } from 'react-reactive-form';

import Add from './index';

describe('<Add />', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      id: '1',
      subcategoryList: ['test'],
      onRequestCategoryUpdate: jest.fn()
    };

    wrapper = shallow(
      <Add {...props} />
    );
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should contain the FieldGroup', () => {
    expect(wrapper.find(FieldGroup)).toHaveLength(1);
  });

  describe('FieldGroup', () => {
    let renderedFormGroup;

    beforeEach(() => {
    });

    describe('rendering', () => {
      it('should render FormGroup correctly', () => {
        renderedFormGroup = (wrapper.find(FieldGroup).shallow().dive());
        expect(renderedFormGroup).toMatchSnapshot();
      });

      it('should render loading', () => {
        wrapper.setProps({ loading: true });

        renderedFormGroup = (wrapper.find(FieldGroup).shallow().dive());
        expect(renderedFormGroup).toMatchSnapshot();
      });
    });

    it('should disable the submit button when no category is selected', () => {
      renderedFormGroup = (wrapper.find(FieldGroup).shallow().dive());
      expect(renderedFormGroup.find('button').prop('disabled')).toBe(true);
    });

    it('should enable the submit button when a  category is selected', () => {
      const form = wrapper.instance().categoryForm;
      const formValue = { sub_category: 'test' };
      form.patchValue(formValue);
      expect(form.value.sub_category).toEqual(formValue.sub_category);
      renderedFormGroup = (wrapper.find(FieldGroup).shallow().dive());
      expect(renderedFormGroup.find('button').prop('disabled')).toBe(false);
    });

    it('should call category update when the form is submitted (search button is clicked)', () => {
      const form = wrapper.instance().categoryForm;
      const formValue = {
        ...form.value,
        sub_category: 'test'
      };
      form.setValue(formValue);
      expect(form.value.sub_category).toEqual(formValue.sub_category);

      renderedFormGroup = (wrapper.find(FieldGroup).shallow().dive());
      // click on the submit button doesn't work in Enzyme, this is the way to test submit functionality
      renderedFormGroup.find('form').simulate('submit', { preventDefault() { } });
      const formCallValue = { ...formValue, _signal: props.id };
      expect(props.onRequestCategoryUpdate).toHaveBeenCalledWith(formCallValue);
    });
  });
});
