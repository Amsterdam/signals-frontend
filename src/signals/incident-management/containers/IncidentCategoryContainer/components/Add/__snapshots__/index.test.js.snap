// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Add /> should render correctly 1`] = `
<div
  className="incident-category-add"
>
  <div
    className="incident-category-add__body"
  >
    <FieldGroup
      control={
        FormGroup {
          "_onDisabledChange": Array [],
          "_pendingChange": false,
          "_pendingDirty": false,
          "_pendingTouched": false,
          "asyncValidator": null,
          "controls": Object {
            "_signal": FormControl {
              "_onCollectionChange": [Function],
              "_onDisabledChange": Array [],
              "_parent": [Circular],
              "_pendingChange": true,
              "_pendingDirty": false,
              "_pendingTouched": false,
              "_pendingValue": "",
              "asyncValidator": null,
              "errors": null,
              "formState": "",
              "get": [Function],
              "getError": [Function],
              "handler": [Function],
              "hasError": [Function],
              "meta": Object {},
              "onBlur": [Function],
              "onBlurChanges": Observable {
                "observers": Array [],
              },
              "onChange": [Function],
              "onValueChanges": Observable {
                "observers": Array [],
              },
              "patchValue": [Function],
              "pristine": true,
              "reset": [Function],
              "setValue": [Function],
              "stateChanges": Observable {
                "observers": Array [],
              },
              "status": "VALID",
              "statusChanges": Observable {
                "observers": Array [],
              },
              "submitted": false,
              "touched": false,
              "validator": null,
              "validatorsOrOpts": Object {
                "asyncValidators": null,
                "updateOn": null,
                "validators": null,
              },
              "value": "",
              "valueChanges": Observable {
                "observers": Array [],
              },
            },
            "loading": FormControl {
              "_onCollectionChange": [Function],
              "_onDisabledChange": Array [],
              "_parent": [Circular],
              "_pendingChange": true,
              "_pendingDirty": false,
              "_pendingTouched": false,
              "_pendingValue": false,
              "asyncValidator": null,
              "errors": null,
              "formState": false,
              "get": [Function],
              "getError": [Function],
              "handler": [Function],
              "hasError": [Function],
              "meta": Object {},
              "onBlur": [Function],
              "onBlurChanges": Observable {
                "observers": Array [],
              },
              "onChange": [Function],
              "onValueChanges": Observable {
                "observers": Array [],
              },
              "patchValue": [Function],
              "pristine": true,
              "reset": [Function],
              "setValue": [Function],
              "stateChanges": Observable {
                "observers": Array [],
              },
              "status": "VALID",
              "statusChanges": Observable {
                "observers": Array [],
              },
              "submitted": false,
              "touched": false,
              "validator": null,
              "validatorsOrOpts": Object {
                "asyncValidators": undefined,
                "updateOn": undefined,
                "validators": undefined,
              },
              "value": false,
              "valueChanges": Observable {
                "observers": Array [],
              },
            },
            "sub": FormControl {
              "_onCollectionChange": [Function],
              "_onDisabledChange": Array [],
              "_parent": [Circular],
              "_pendingChange": true,
              "_pendingDirty": false,
              "_pendingTouched": false,
              "_pendingValue": "",
              "asyncValidator": null,
              "errors": Object {
                "required": true,
              },
              "formState": "",
              "get": [Function],
              "getError": [Function],
              "handler": [Function],
              "hasError": [Function],
              "meta": Object {},
              "onBlur": [Function],
              "onBlurChanges": Observable {
                "observers": Array [],
              },
              "onChange": [Function],
              "onValueChanges": Observable {
                "observers": Array [],
              },
              "patchValue": [Function],
              "pristine": true,
              "reset": [Function],
              "setValue": [Function],
              "stateChanges": Observable {
                "observers": Array [],
              },
              "status": "INVALID",
              "statusChanges": Observable {
                "observers": Array [],
              },
              "submitted": false,
              "touched": false,
              "validator": [Function],
              "validatorsOrOpts": Object {
                "asyncValidators": null,
                "updateOn": null,
                "validators": [Function],
              },
              "value": "",
              "valueChanges": Observable {
                "observers": Array [],
              },
            },
          },
          "errors": null,
          "get": [Function],
          "getError": [Function],
          "handleSubmit": [Function],
          "hasError": [Function],
          "meta": Object {},
          "patchValue": [Function],
          "pristine": true,
          "reset": [Function],
          "setValue": [Function],
          "stateChanges": Observable {
            "observers": Array [],
          },
          "status": "INVALID",
          "statusChanges": Observable {
            "observers": Array [],
          },
          "submitted": false,
          "touched": false,
          "validator": null,
          "validatorOrOpts": Object {
            "asyncValidators": null,
            "updateOn": null,
            "validators": null,
          },
          "value": Object {
            "_signal": "",
            "loading": false,
            "sub": "",
          },
          "valueChanges": Observable {
            "observers": Array [],
          },
        }
      }
      render={[Function]}
      strict={true}
    />
  </div>
</div>
`;
